src/jit/lsra.cpp

```diff
@@ -5062,26 +5107,70 @@ void LinearScan::freeRegister(RegRecord* physRegRecord)
     }
 }
 
-void LinearScan::freeRegisters(regMaskTP regsToFree)
+//------------------------------------------------------------------------
+// LinearScan::updateRegs: Update register after allocating a RefPosition
+//
+// Arguments:
+//    currentInterval    - the Interval for the RefPositoin
+//    currentRefPosition - the RefPosition that has been allocated
+//
+// Return Value:
+//    True iff the register has been unassigned.
+//
+void LinearScan::RegsToFree::updateRegs(Interval* interval, RefPosition* refPosition, regMaskTP regMask)
 {
-    if (regsToFree == RBM_NONE)
+    bool         unassign        = false;
+    RefPosition* nextRefPosition = refPosition->nextRefPosition;
+    if (refPosition->lastUse || nextRefPosition == nullptr)
     {
-        return;
+        assert(refPosition->isIntervalRef());
+
+        if (refPosition->copyReg || (refPosition->refType != RefTypeExpUse && nextRefPosition == nullptr))
+        {
+            unassign = true;
+        }
+        else
+        {
+            interval->isActive = false;
+        }
     }
+    if (unassign)
+    {
+        if (refPosition->delayRegFree)
+        {
+            delayed |= regMask;
+        }
+        else
+        {
+            current |= regMask;
+        }
+    }
+}
 
+//------------------------------------------------------------------------
+// LinearScan::freeRegisters: Free the registers in 'regsToFree'
+//
+// Arguments:
+//    regsToFree         - the 'RegsToFree' struct
+//
+void LinearScan::freeRegisters(RegsToFree& regsToFree)
```

Comment:
Our code convention forbids passing by references. 
I have spent a few seconds trying to understand where regs 5261 `ToFree         = RBM_NONE;` disappeared.

Reply:
Changed to take a pointer.